src/player_hardcoded_effects.cpp

```diff
@@ -1336,14 +1336,15 @@ void player::hardcoded_effects( effect &it )
                         // 10 minute cyber-snooze
                         it.mod_duration( 100 );
                     }
-                } else {
+                }
+            } else {
+                if( dur <= 1 ) {
+                    add_msg_if_player( m_warning, _( "Your alarm clock rings, but you do not hear it." ) );
```

Comment:
Please remove this, we don't notify the player of things they don't percieve.
I could maybe see an argument for it *sometimes* notifying the player that they shut off the alarm but fall back asleep, but that seems like it would be a bit contentious, so it needs to be in another PR.